### PR TITLE
Add new API to set pop attestation flags and retrieve chain

### PR DESCRIPTION
**Pull request by iambmelt**
This is preemptive work to support exposing the certificate chain of the device pop key

Note:
- PoP keys initialized _before_ this PR won't have a certificate chain (will return `null` or `1` containing self) -- I'm determining this _not_ to be something I need to account for in this PR, given there is no prod deployment/adoption of these APIs yet

### CODE REVIEW

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/DevicePoPManagerTests.java [hunk 1]
```diff
@@ -449,4 +451,22 @@ public void testGenerateShr() throws ClientException, MalformedURLException, Par
         Assert.assertNotNull(jwtClaimsSet.getClaim("cnf"));
         Assert.assertEquals(clientClaims, jwtClaimsSet.getClaim("client_claims"));
     }
+
+    @Test
+    public void testHasCertificateChain24() throws ClientException {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/internal/platform/DevicePoPManagerTests.java:457]: Can you annotate this method and have that work?

File: common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java [hunk 2]
```diff
@@ -1307,6 +1340,14 @@ private void initialize23(@NonNull final KeyPairGenerator keyPairGenerator,
         keyPairGenerator.initialize(spec);
     }
 
+    @SuppressLint("NewApi")
+    @RequiresApi(Build.VERSION_CODES.N)
+    @NonNull
+    private KeyGenParameterSpec.Builder setAttestationChallenge(
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java:1346]: Can we add a test for these?

**Review by AdamBJohnsonx — COMMENTED** [review]
I feel like we should have a test.

File: common/src/androidTest/java/com/microsoft/identity/common/internal/platform/DevicePoPManagerTests.java (hunk 1 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/internal/platform/DevicePoPManagerTests.java]: Annotated - I can revert it if this isn't respected by the test runner

File: common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java (hunk 2 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java:1346]: The `getCertificateChain()` is the test -- [that function won't return a result unless an attestation challenge is set](https://developer.android.com/reference/android/security/keystore/KeyGenParameterSpec.Builder#setAttestationChallenge(byte[])) -- if there's some other aspect of this we can test, I'm open to ideas. There didn't seem to be a lot to do here

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java:1346]: Yeah , this is private.  Oops.

**Review by AdamBJohnsonx — APPROVED** [review]

**Review by joakley-msft — APPROVED** [review]

**Pull request merged**